src/Routes/MoviePage/index.js

```diff
@@ -0,0 +1,68 @@
+import { MovieInfo, Cast, Crew, CastHeader, CrewHeader, CastList, CrewList } from "./styled";
+import MovieCardFull from "../../common/MovieCardFull"
+import MovieHeading from "./components/MovieHeading.js"
+import { useParams } from "react-router-dom";
+import { useDataById } from "../../useDataById";
+import { useSelector } from "react-redux";
+import { selectMovies, selectCast, selectCrew } from "../../moviesSlice";
+import { useMovieCast } from "../../useMovieCast";
+import PersonCard from "../../common/PersonCard";
+
+const MoviePage = () => {
+    const params = useParams();
+
+    useDataById(params.id);
+    useMovieCast(params.id);
+
+    const movie = useSelector(selectMovies);
+    const cast = useSelector(selectCast);
+    const crew = useSelector(selectCrew);
+
+    // these if statements are kinda bad but i didnt know any other way of making operations on selected variables :C
+
+    let rating = 0;
```

Comment:
we can consider somethink like that " let rating = movie.vote_average ? movie.vote_average.toFixed(1) : 0; "

Reply:
the problem is i cant make any operations without a if statement it seems, and definitely not instantly since the component loads asynchronously from the data